app/models/manageiq/providers/embedded_terraform/automation_manager/job.rb

```diff
@@ -138,4 +146,18 @@ def cleanup_git_repository
   rescue Errno::ENOENT
     nil
   end
+
+  def update_service_instance_with_stack_id(action, service_instance_id, stack_id)
+    if !action.nil? && action.downcase == 'provision'
+      service_instance = ServiceTerraformTemplate.find_by(:id => service_instance_id)
+      if !service_instance.nil?
+        service_instance.preprocess(action, {:extra_vars => {:miq_terraform_stack_id => stack_id}})
+        _log.debug("updated service-instance/#{service_instance.to_json} with stack_id#{stack_id}")
+      else
+        _log.info("Not found service-instance/#{service_instance_id}")
+      end
+    else
+      _log.debug("Not Provision action, no update to service-instance/#{service_instance_id} required")
+    end
+  end
```

Comment:
Believe we discussed this in chat, but just to make sure this doesn't get lost I'll add here also.

I think the proper way to do this is to have the service retirement retire all of the related resources which in this case will be an OrchestrationStack.  This would also work if in the future you had a service that ends up creating multiple terraform_stacks where what you have here would preclude that.

Reply:
@agrare 
The `update_service_instance_with_stack_id()` method is no longer required,  has been removed. We now fetch `stack_id` from the Provision Job,  where it available, and use it for Retirement.

Terraform-Runner as reference to all the created resources from the Provision Job,  all it needs the relevant Terraform-Tunner `stack_id` to Retire ( `terraform destroy` ) the resources.